app/services/control_rate_service.rb

```diff
@@ -46,17 +46,25 @@ def call
 
       data[:registrations] = registration_counts
       data[:cumulative_registrations] = sum_cumulative_registrations
+      data[:registrations].delete_if { |period, value| !periods.cover?(period) }
+      data[:cumulative_registrations].delete_if { |period, value| !periods.cover?(period) }
```

Comment:
This removes registration data that _is not_ part of the range requested. If we didn't do this calling code might end up displaying too much data in reports or charts.